datafusion/functions/src/datetime/mod.rs

```diff
@@ -129,11 +151,22 @@ pub mod expr_fn {
     pub fn to_timestamp_nanos(args: Vec<Expr>) -> Expr {
         super::to_timestamp_nanos().call(args)
     }
+
+    pub fn _date_bin_scalar_udf() -> Arc<ScalarUDF> {
```

Comment:
I'm not sure happy with having these functions here but I could see no other way to provide access to the ScalarUDF